test/e2e/storage/testsuites/base.go

```diff
@@ -95,16 +95,50 @@ var CSISuites = append(BaseSuites,
 	InitSnapshottableStressTestSuite,
 )
 
-// TestSuite represents an interface for a set of tests which works with TestDriver
-type TestSuite interface {
-	// GetTestSuiteInfo returns the TestSuiteInfo for this TestSuite
-	GetTestSuiteInfo() TestSuiteInfo
-	// DefineTests defines tests of the testpattern for the driver.
+// TestSuiteInterface represents an interface for a set of tests which works with TestDriver.
+// Each testsuite should implement this interface and use a TestSuiteHandler to define its Tests.
+// The function defined inside should be called only by TestSuiteHandler.
+// All the functions are designed to not be exported so that caller has to use TestSuiteHandler to
+// actually define tests.
+type TestSuiteInterface interface {
+	getTestSuiteInfo() TestSuiteInfo
+	// defineTests defines tests of the testpattern for the driver.
 	// Called inside a Ginkgo context that reflects the current driver and test pattern,
 	// so the test suite can define tests directly with ginkgo.It.
-	DefineTests(TestDriver, testpatterns.TestPattern)
-	// SkipRedundantSuite will skip the test suite based on the given TestPattern and TestDriver
-	SkipRedundantSuite(TestDriver, testpatterns.TestPattern)
+	defineTests(TestDriver, testpatterns.TestPattern)
+	// skipUnsupportedTests will skip the test suite based on the given TestPattern, TestDriver
+	// Testsuite should check if the given pattern and driver works for the "whole testsuite"
+	// Testcase specific check should happen inside defineTests
+	skipUnsupportedTests(TestDriver, testpatterns.TestPattern)
+}
+
+// TestSuiteHandler is the actual data struct we use to define tests
+// All the test registration should happen from the handler
+type TestSuiteHandler struct {
```

Comment:
From my understanding, each test suite constructor now returns a `TestSuiteHandler`, and these handlers are responsible for registering their test suite's tests (which skips certain driver and pattern combinations).

Does this `TestSuiteHandler` need to be tied to the test suite? It's not clear to me what advantage that provides, vs. keeping it separate from the test suite. All of the information we need for registering and defining tests comes from the test suite field, which suggests to me that this wrapper struct may not be necessary.